### PR TITLE
conditional case claim

### PR DESCRIPTION
**Pull request by proteusvacuum**
This adds the ability to provide a custom relevant condition for case claim. 
FYI @kaapstorm 
Code buddy @dannyroberts ? or @millerdev can't tell.

### CODE REVIEW

File: corehq/apps/app_manager/views/modules.py [hunk 1]
```diff
@@ -697,10 +699,19 @@ def edit_module_detail_screens(request, domain, app_id, module_id):
...
     if search_properties is not None:
-        module.search_config = CaseSearch(properties=[
-            CaseSearchProperty.wrap(p) for p in _update_search_properties(module, search_properties, lang)
-        ])
-        # TODO: Add UI and controller support for CaseSearch.command_label
+        if search_properties.get('properties') is not None:
+            module.search_config = CaseSearch(
+                properties=[CaseSearchProperty.wrap(p)
+                            for p in _update_search_properties(
+                                    module,
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/app_manager/views/modules.py [hunk 2]
```diff
@@ -697,10 +699,19 @@ def edit_module_detail_screens(request, domain, app_id, module_id):
...
+                                    search_properties.get('properties'), lang
+                            )],
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/app_manager/views/modules.py [hunk 3]
```diff
@@ -697,10 +699,19 @@ def edit_module_detail_screens(request, domain, app_id, module_id):
...
-            CaseSearchProperty.wrap(p) for p in _update_search_properties(module, search_properties, lang)
-        ])
-        # TODO: Add UI and controller support for CaseSearch.command_label
+        if search_properties.get('properties') is not None:
+            module.search_config = CaseSearch(
+                properties=[CaseSearchProperty.wrap(p)
+                            for p in _update_search_properties(
+                                module,
+                                search_properties.get('properties'), lang
+                            )],
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/app_manager/views/modules.py [hunk 4]
```diff
@@ -697,10 +699,21 @@ def edit_module_detail_screens(request, domain, app_id, module_id):
...
-        module.search_config = CaseSearch(properties=[
-            CaseSearchProperty.wrap(p) for p in _update_search_properties(module, search_properties, lang)
-        ])
-        # TODO: Add UI and controller support for CaseSearch.command_label
+        if search_properties.get('properties') is not None:
+            module.search_config = CaseSearch(
+                properties=[
+                    CaseSearchProperty.wrap(p)
+                    for p in _update_search_properties(
+                            module,
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: E126 continuation line over-indented for hanging indent

**Comment by millerdev** [comment]
test failure looks relevant (pun intended)
```
FAIL: Suite should include sync-request if searching is configured
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_suite_sync_request.py", line 35, in test_sync_request
    self.assertXmlPartialEqual(self.get_xml('sync_request'), suite, "./sync-request[1]")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 26, in assertXmlPartialEqual
    self.assertXmlEqual(expected, actual, normalize=False)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 126, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -1,6 +1,6 @@
 <partial>
   <sync-request>
-    <post relevant="&#10;count(instance('casedb')/casedb/case[@case_id=instance('querysession')/session/data/case_id]) = 0&#10;" url="https://www.example.com/a/test_domain/phone/claim-case/">
+    <post relevant="count(instance('casedb')/casedb/case[@case_id=instance('querysession')/session/data/case_id]) = 0" url="https://www.example.com/a/test_domain/phone/claim-case/">
       <data key="case_id" ref="instance('querysession')/session/data/case_id"/>
     </post>
     <command id="search_command.m0">
```

**Comment by kaapstorm** [comment]
Cool that this is now manageable from the UI.

File: corehq/apps/app_manager/const.py [hunk 5]
```diff
@@ -53,3 +53,5 @@
 AMPLIFIES_NOT_SET = 'not_set'
 
 DEFAULT_MONTH_FILTER_PERIOD_LENGTH = 0
+
+CLAIM_DEFAULT_RELEVANT_CONDITION = "count(instance('casedb')/casedb/case[@case_id=instance('querysession')/session/data/case_id]) = 0"
```
Comment (dimagimon) [inline corehq/apps/app_manager/const.py]: E501 line too long (134 > 115 characters)

**Comment by proteusvacuum** [comment]
:white_check_mark:

**Pull request merged**